applications/ATOM/train_atom_char_rnn.py

```diff
@@ -40,133 +84,194 @@ def construct_model():
-        loss.append(lbann.LayerTerm(ce_mask, scale=1/(sequence_length-1)))
-
+        y_soft = lbann.Softmax(fc_l, name="soft_" + str(i))
+        gt = lbann.OneHot(idl[i + 1], size=num_embeddings)
+        ce = lbann.CrossEntropy([y_soft, gt], name="loss_" + str(i))
+        # mask padding in input
+        pad_mask = lbann.NotEqual(
+            [idl[i], lbann.Constant(value=run_args.pad_index, num_neurons="1")],
+        )  
+        ce_mask = lbann.Multiply([pad_mask, ce], name="loss_mask_" + str(i))
+        loss.append(lbann.LayerTerm(ce_mask, scale=1 / (sequence_length - 1)))
 
     layers = list(lbann.traverse_layer_graph(input))
     # Setup objective function
     weights = set()
     for l in layers:
-      weights.update(l.weights)
+        weights.update(l.weights)
     obj = lbann.ObjectiveFunction(loss)
 
-
-    callbacks = [lbann.CallbackPrint(),
-                 lbann.CallbackTimer(),
-                 lbann.CallbackStepLearningRate(step=10, amt=0.5),
-                 lbann.CallbackDumpWeights(basename="weights",epoch_interval=50)]
+    callbacks = [
+        lbann.CallbackPrint(),
+        lbann.CallbackTimer(),
+        lbann.CallbackStepLearningRate(step=run_args.step_size, amt=run_args.gamma),
+        lbann.CallbackDumpWeights(basename=run_args.dump_weights_dir, epoch_interval=1),
+    ]
 
     # Construct model
-    mini_batch_size = 64
-    num_epochs = 50
-    return lbann.Model(mini_batch_size,
-                       num_epochs,
-                       weights=weights,
-                       layers=layers,
-                       objective_function=obj,
-                       callbacks=callbacks)
-
-def construct_data_reader():
-    """Construct Protobuf message for Python data reader.
+    return lbann.Model(
+        run_args.batch_size,
+        run_args.num_epochs,
+        weights=weights,
+        layers=layers,
+        objective_function=obj,
+        callbacks=callbacks,
+    )
+
+
+def construct_data_reader(run_args):
+    """
+    Construct Protobuf message for Python data reader.
 
     The Python data reader will import this Python file to access the
     sample access functions.
 
     """
+
     import os.path
     import lbann
-    module_file = os.path.abspath(__file__)
+
+    module_file = os.path.abspath(run_args.data_module_file)
+    os.environ["DATA_CONFIG"] = os.path.abspath(run_args.data_config)
+
     module_name = os.path.splitext(os.path.basename(module_file))[0]
     module_dir = os.path.dirname(module_file)
 
+    print("module_name: {}\tmodule_dir: {}".format(module_name, module_dir))
+
     # Base data reader message
     message = lbann.reader_pb2.DataReader()
 
     # Training set data reader
     data_reader = message.reader.add()
-    data_reader.name = 'python'
-    data_reader.role = 'train'
+    data_reader.name = "python"
+    data_reader.role = "train"
     data_reader.shuffle = True
-    data_reader.percent_of_data_to_use = 1.0
+    data_reader.percent_of_data_to_use = 0.001
```

Comment:
Is it intentional to use 0.1% of the data?

Reply:
I changed it back to 1.0